samples/sensor/a01nyub/src/main.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 SteadConnect
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+
+LOG_MODULE_REGISTER(main);
+
+int main(void)
+{
+	LOG_INF("Booted DFRobot A01NYUB distance sensor sample");
+
+	const struct device *dev;
+	struct sensor_value val;
+
+	dev = DEVICE_DT_GET_ONE(dfrobot_a01nyub);
+
+	if (!device_is_ready(dev)) {
+		LOG_ERR("sensor: device not found.");
+		return 0;
+	}
+
+	while (1) {
+		if (sensor_sample_fetch(dev) != 0) {
+			LOG_ERR("sensor: sample fetch fail.");
+		}
+
+		if (sensor_channel_get(dev, SENSOR_CHAN_DISTANCE, &val) != 0) {
```

Comment:
We're trying to reduce the number of driver-specific samples. Please abstract this sample to allow support for other `SENSOR_CHAN_DISTANCE` sensor drivers. See `samples/sensor/*_polling` for example

Reply:
That's good guidance, thanks. In this case, the sample doesn't add anything that is not already covered by the mhz19b sample, which is another interrupt-driven UART-based sensor. I have removed it from the PR.